<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogGlobal</class>
 <widget class="QDialog" name="DialogGlobal">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>758</width>
    <height>782</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_10">
   <item row="5" column="0" colspan="2">
    <widget class="QWidget" name="widget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="pushButtonProperties">
        <property name="text">
         <string>Properties...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonDefaults">
        <property name="text">
         <string>Defaults</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>518</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonOK">
        <property name="text">
         <string>OK</string>
        </property>
        <property name="default">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonCancel">
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Switches</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QCheckBox" name="checkBoxAllowConnectedCollisions">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="text">
         <string>Allow Connected Collisions</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="checkBoxAllowInternalCollisions">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="text">
         <string>Allow Internal Collisions</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="checkBoxSpringDamping">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="text">
         <string>Spring &amp;&amp; Damping Constant Mode</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="groupBox_7">
     <property name="title">
      <string>Limits</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_8">
      <item row="0" column="2">
       <widget class="LineEditDouble" name="lineEditTimeLimit">
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_20">
        <property name="text">
         <string>Mechanical Energy</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="LineEditDouble" name="lineEditMetabolicEnergyLimit"/>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_21">
        <property name="text">
         <string>Time</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="LineEditDouble" name="lineEditMechanicalEnergyLimit"/>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_19">
        <property name="text">
         <string>Metabolic Energy</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Gravity</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="0" column="1">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>X</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="LineEditDouble" name="lineEditGravityX">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Y</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="LineEditDouble" name="lineEditGravityY">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Z</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="LineEditDouble" name="lineEditGravityZ">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <spacer name="horizontalSpacer_8">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QGroupBox" name="groupBoxWarehouse">
     <property name="title">
      <string>Warehouse</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="3" column="2">
       <widget class="LineEditDouble" name="lineEditUnitIncreaseDistanceThreshold"/>
      </item>
      <item row="1" column="2">
       <widget class="LineEditDouble" name="lineEditWarehouseDecreaseThresholdFactor"/>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Decrease Threshold</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="LineEditPath" name="lineEditCurrentWarehouseFile">
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Unit Increase</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Fail Distance</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>File</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="LineEditDouble" name="lineEditFailDistanceAbort"/>
      </item>
      <item row="0" column="0">
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Select</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_9">
      <item row="0" column="2">
       <widget class="QComboBox" name="comboBoxFitnessType">
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Fitness Type</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="2">
       <widget class="QComboBox" name="comboBoxDistanceTravelledBodyIDName"/>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Reference Body</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox_6">
     <property name="title">
      <string>Integrator</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="2" column="1">
       <widget class="QLabel" name="labelCFM">
        <property name="text">
         <string>CFM</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QComboBox" name="comboBoxStepType">
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Integration engine to use</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="labelERP">
        <property name="text">
         <string>ERP</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="LineEditDouble" name="lineEditCFM">
        <property name="toolTip">
         <string>Constraint Force Mixing parameter (typically 1e-10)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="3" column="2">
       <widget class="LineEditDouble" name="lineEditERP">
        <property name="toolTip">
         <string>Error Reduction Parameter (typically 0.1 for soft and 0.8 for hard simulations)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Step Type</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Step Size</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="LineEditDouble" name="lineEditStepSize">
        <property name="toolTip">
         <string>Integration step size (typical values 1e-3 to 1e-5)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Mesh Path</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QListWidget" name="listWidgetMeshPath">
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QGroupBox" name="groupBox_8">
     <property name="title">
      <string>Extras</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_7">
      <item row="0" column="0">
       <spacer name="horizontalSpacer_9">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Linear Damping</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="LineEditDouble" name="lineEditLinearDamping">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Angular Damping</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="LineEditDouble" name="lineEditAngularDamping">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Error Score</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="LineEditDouble" name="lineEditNumericalErrorScore">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Permitted Errors</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QSpinBox" name="spinBoxPermittedErrorCount">
        <property name="minimum">
         <number>-1</number>
        </property>
        <property name="maximum">
         <number>999999</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="groupBox_9">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Contact Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_6">
      <item row="0" column="0">
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>16</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>Max Correcting Velocity</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="LineEditDouble" name="lineEditContactMaxCorrectingVel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>201</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Surface Layer Thickness</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="LineEditDouble" name="lineEditContactSurfaceLayer">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LineEditDouble</class>
   <extends>QLineEdit</extends>
   <header>LineEditDouble.h</header>
  </customwidget>
  <customwidget>
   <class>LineEditPath</class>
   <extends>QLineEdit</extends>
   <header>LineEditPath.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>comboBoxStepType</tabstop>
  <tabstop>lineEditCFM</tabstop>
  <tabstop>lineEditERP</tabstop>
  <tabstop>checkBoxAllowConnectedCollisions</tabstop>
  <tabstop>checkBoxAllowInternalCollisions</tabstop>
  <tabstop>comboBoxFitnessType</tabstop>
  <tabstop>comboBoxDistanceTravelledBodyIDName</tabstop>
  <tabstop>lineEditContactMaxCorrectingVel</tabstop>
  <tabstop>lineEditContactSurfaceLayer</tabstop>
  <tabstop>lineEditTimeLimit</tabstop>
  <tabstop>lineEditMechanicalEnergyLimit</tabstop>
  <tabstop>lineEditMetabolicEnergyLimit</tabstop>
  <tabstop>lineEditGravityX</tabstop>
  <tabstop>lineEditGravityY</tabstop>
  <tabstop>lineEditGravityZ</tabstop>
  <tabstop>lineEditCurrentWarehouseFile</tabstop>
  <tabstop>lineEditWarehouseDecreaseThresholdFactor</tabstop>
  <tabstop>lineEditFailDistanceAbort</tabstop>
  <tabstop>lineEditUnitIncreaseDistanceThreshold</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
